sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs

```diff
@@ -414,11 +414,11 @@ public virtual void CancelScheduledMessagesException(string identifier, string e
 
         #region Settlement
         [Event(CompleteMessageStartEvent, Level = EventLevel.Informational, Message = "{0}: CompleteAsync start. MessageCount = {1}, LockTokens = {2}")]
-        public virtual void CompleteMessageStart(string identifier, int messageCount, string lockTokens)
+        public virtual void CompleteMessageStart(string identifier, int messageCount, Guid lockToken)
```

Comment:
I'm not sure that this will work, since the parameter list doesn't match list passed to WriteEvent (and I'm also unsure if Guid is even supported as a parameter to an event). I think what we would need to do (barring confirmation that this approach works by adding live test coverage in EventSourceLiveTests) in order to lazily call ToString would be to add wrapper NonEvent methods, similar to https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/src/Diagnostics/ServiceBusEventSource.cs#L378, that call into the actual Event methods.

Reply:
Ah yeah I remember now. I haven't worked a while with event sources. My knowledge is a bit rusty. Will fix

Reply:
Yeah I'm always a bit wary when dealing with these as when they fail, it is not really surfaced unless you are inspecting the logs.

Reply:
Have a look. Hopefully I got it right this time

Reply:
Looks good!